docs/documentation/repository-pattern.md

```diff
@@ -82,3 +93,99 @@ the repository controls.
 
 
 ## Examples
+
+The following code-snippets are examples to get a bet unterstanding of the repository pattern. Furthermore, these examples can be used as templates for the developement of new features and refactoring of already existing ones.
+
+Note that this examples are not perfect for every situation. Like always in software developemt, the developer has to decide which is the best way of usage for his or hers appliaction.
+
+For simplicity reasons, the class of the objects to persist is in all following example the same (except for some additions that are mentioned later). The class is called `ilObjGeoLocation` and is immutble:
+
+    public class ilObjGeoLocation() {
```

Comment:
Use ` ```php ` to start and ` ``` ` to end a code block. GitHub will take care of syntaxhighlighting then.

Reply:
Thank you for this markdown trick! :-) Changed it with latest commit.